src/openrct2/world/Sprite.cpp

```diff
@@ -884,11 +889,10 @@ uint16_t remove_floating_sprites()
  */
 static bool sprite_should_tween(rct_sprite* sprite)
 {
-    switch (sprite->generic.linked_list_index)
+    switch (sprite->generic.sprite_identifier)
     {
-        case SPRITE_LIST_PEEP:
-        case SPRITE_LIST_TRAIN_HEAD:
-        case SPRITE_LIST_VEHICLE:
+        case SPRITE_IDENTIFIER_PEEP:
+        case SPRITE_IDENTIFIER_VEHICLE:
             return true;
     }
     return false;
```

Comment:
This function could also do `return sprite->IsPeep() || sprite->IsVehicle()`.

Reply:
we don't have an `isVehicle()` function yet. It will be changed as part of the is / as refactor.